src/inference/rejection.js

```diff
@@ -33,28 +41,49 @@ module.exports = function(env) {
     this.oldCoroutine = env.coroutine;
     env.coroutine = this;
 
-    if (!_.isNumber(this.numSamples) || this.numSamples <= 0) {
-      throw new Error('samples should be a positive integer.');
-    }
-
     if (this.incremental && this.maxScore > 0) {
       util.warn('Rejection: Reduce maxScore to zero for better performance.');
     }
   }
 
   Rejection.prototype.run = function() {
+    if (!_.isNumber(this.numSamples) || this.numSamples <= 0) {
+      return this.error('"samples" should be a positive integer.');
+    }
+    var elapseSec = (Date.now() - this.startTime) / 1000.0;
+    if (elapseSec > 2) {
```

Comment:
`startTime` isn't reset on each iteration, so it looks like `elapseSec > 2` will always be true after it has been true once and `elapseSec` will increase monotonically. Did you intend to divide by `elapseSec` instead of `2` below?